### ISSUE TITLE
[New command] Test-DbaScheduledTask, 

### ISSUE THREAD

**Issue by MARCO-K** [sqlcollaborative/dbatools#828]
﻿### Is this a *feature* OR *bug*:
This is a new feature.

### Details
Basically the cmdlet verifies if ScheduledTask state is configured accorind the Windows-based best practice.

The intention of this new command is to verify the state of the scheduled taks "ScheduledDefrag". I had huge performance hits causing SQL iserver to slow down a lot with the task enabled.

**Comment by ctrlbold** [comment]
I'm intrigued :) How would we test that the scheduled tasks state is best practice?

**Comment by MARCO-K** [comment]
Hi,

it's simply comparing:
$bpState = 'Disabled'
$taskname = 'ScheduledDefrag'
$task = Get-ScheduledTask | Where-Object Taskname -eq $taskname
$task.State -eq $bpState

If already created the script and doing some testing now.

**Comment by ctrlbold** [comment]
I'm digging it. Any others on that list?

Remember, though, that we write for PS 3 which does not include Get-ScheduledTask.

You'll have to use `New-Object -ComObject Schedule.Service` or something similar

**Comment by MARCO-K** [comment]
jsut started with the defrag because i had performance problems quite recently.

Thanks. for the advice. I'll change it in my script.

**Comment by wsmelton** [comment]
The only issue with this is that task is not a standard for Windows OS. It is also not particularly best practice to just run disk defrag on a disk that SQL Server is using, unless you follow [Microsoft guidelines](https://support.microsoft.com/en-us/help/3195161/defragmenting-sql-server-database-disk-drives)

Basically, you will never find a scheduled task on a given server called "ScheduledDefrag" or anything else because it is not there by default. The only way that task exist is if you create it with your command, or you happen across an environment that someone created it with the exact same name. So looking for that name specifically is not going to work.

Final point, generally speaking on systems that have been in production for months or years and the database files have grown exponentially you are not going to see additional performance by running defrag. If you do it is generally not going to be worth the trouble of taking the outage window required for it. [More info here](https://blogs.technet.microsoft.com/askcore/2014/02/17/whats-new-in-defrag-for-windows-server-20122012r2/).

**Comment by ctrlbold** [comment]
I had checked this and it is there by default -- this is on my fresh 2016 box

![image](https://cloud.githubusercontent.com/assets/8278033/23309306/79dbe6d2-faae-11e6-84ee-fdc5a3bd8c0f.png)

**Comment by MARCO-K** [comment]
In fact this jobs is enabled by default since 2012. That was cause a lot of trouble for me.

**Comment by niphlod** [comment]
there's more to it than it seems. 
https://blogs.technet.microsoft.com/askcore/2014/02/17/whats-new-in-defrag-for-windows-server-20122012r2/

**Comment by niphlod** [comment]
And suggesting disabling for everyone as a defacto standard isn't a smart choice... defragging still needs to be done to avoid slowly crawling performances.

**Comment by niphlod** [comment]
tl;dr: run defrag.exe -c -h -k -g -$ and see it exiting instantaneously almost everywhere.

**Comment by wsmelton** [comment]
This is strictly my opinion, but I have been in multiple high-end environments that were doing millions of transactions a second for online sites. I have never had to disable this scheduled task, and likely would never do so. 

That task itself is hidden, you cannot visibly access it using Task Scheduler and the only way to even disable it that I can find is from command line. 

In the years I have been support SQL Server since 2012 OS came out I have never come across this scheduled task being an issue. I also do not believe it is best practice to disable that task, MS put it there for a reason. If you believe it is best practice then you need to come up with an authoritative reference stating as such. In that sense I would not make any mention of this being best practice unless it can be shown that it is a recommended practice from either MS or the community.

**Comment by niphlod** [comment]
going to close this in a few days if @MARCO-K got what we are telling (which is, basically, that task isn't definitely a bad practice)

**Comment by potatoqualitee** [comment]
agreed, thanks you guys for the feedback and taking the time to explain!
